core/src/main/java/org/mule/runtime/core/privileged/routing/CompositeRoutingException.java

```diff
@@ -54,16 +64,46 @@ public CompositeRoutingException(RoutingResult routingResult) {
   @Override
   public String getDetailedMessage() {
     StringBuilder builder = new StringBuilder();
-    builder.append(MESSAGE_TITLE).append(lineSeparator());
-
-    for (Entry<String, Error> entry : routingResult.getFailures().entrySet()) {
-      String routeSubtitle = String.format("Route %s: ", entry.getKey());
-      MuleException muleException = ExceptionHelper.getRootMuleException(entry.getValue().getCause());
-      if (muleException != null) {
-        builder.append(routeSubtitle).append(muleException.getDetailedMessage());
-      } else {
-        builder.append(routeSubtitle)
-            .append("Caught exception in Exception Strategy: " + entry.getValue().getCause().getMessage());
+    // provide information about the composite exception itself
+    builder.append(super.getDetailedMessage());
+
+    // get detailed information about the exceptions that make up the composite exception
+    builder.append(lineSeparator()).append(MESSAGE_SUB_TITLE).append(lineSeparator());
+
+    Method getDetailedFailuresMethod = null;
+    Map<String, Pair<Error, EventProcessingException>> detailedFailures = null;
+    try {
+      getDetailedFailuresMethod = RoutingResult.class.getMethod("getFailuresWithExceptionInfo");
+      detailedFailures =
+          (Map<String, Pair<Error, EventProcessingException>>) getDetailedFailuresMethod.invoke(routingResult);
+    } catch (InvocationTargetException | NoSuchMethodException | IllegalAccessException e) {
+      e.printStackTrace();
+      LOGGER.warn("Invalid Invocation, Expected method doesn't exist");
+    }
+
+    if (detailedFailures.isEmpty()) {
+      // If we only have an error map that doesn't provide information about exceptions, process with original logic
+      for (Entry<String, Error> entry : routingResult.getFailures().entrySet()) {
+        String routeSubtitle = String.format("Route %s: ", entry.getKey());
+        MuleException muleException = ExceptionHelper.getRootMuleException(entry.getValue().getCause());
+        if (muleException != null) {
+          builder.append(routeSubtitle).append(muleException.getDetailedMessage());
+        } else {
+          builder.append(routeSubtitle)
+              .append("Caught exception in Exception Strategy: " + entry.getValue().getCause().getMessage());
+        }
+      }
+    } else {
+      // If we are provided an error map that also stores the exceptions that make up the composite exception,
+      // process with new logic to provide detailed error message
+      for (Entry<String, Pair<Error, EventProcessingException>> entry : detailedFailures.entrySet()) {
+        String routeSubtitle = String.format("Route %s: ", entry.getKey());
+        MuleException muleException = entry.getValue().getSecond();
+
+        builder.append(lineSeparator());
+        if (muleException != null) {
+          builder.append(routeSubtitle).append(muleException.getVerboseMessage());
```

Comment:
I used getVerboseMessage() to print the missing information by default(https://docs.google.com/document/d/1GmorZFPPVlCQpDYnDJUyfUh1j4dbToy4gTxaym8YoOg/edit?usp=sharing). I changed it back to getDetailedMessage() so that the customer still can get the missing information using `-M-Dmule.verbose.exceptions=true`.